### ISSUE TITLE
Custom headers disappear when the page is access by 'OPTIONS' method.

### ISSUE THREAD

**Issue by jim-king-2000** [vercel/next.js#19044]
# Bug report

## Describe the bug

Set custom headers to index.js. Then, issue an 'OPTIONS' request to `localhost:3000`, notice that there are no custom headers we just set. It is OK when issuing a 'GET' method.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Set custom headers to index.js.
2. Open postman and issue an 'OPTIONS' request to `localhost:3000`.
3. Watch the response headers. Notice there are no custom headers.
4. Issue a 'GET' request, notice there are custom headers.

## Expected behavior

Provide custom headers in all methods.

## System information

- OS: Windows 10
- Browser (if applies): Chrome 86 & Firefox 82
- Version of Next.js: 10.0.1
- Version of Node.js: 15.2.0

## Additional information
This bug prevent us from cross-domain redirection, which is a *MUST* when using 3rd-party login. The browser would display the following error message:

```
submit:1 Access to fetch at 'https://partner.szesum.com/customLoginPage?uid=sUujKVZrjUm-Xe8Qq_uj-&action=signin&error=Invalid%20username%20or%20password.' (redirected from 'https://partner.insightauth.com/interaction/sUujKVZrjUm-Xe8Qq_uj-/signin') from origin 'https://partner.insightauth.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
submit-dbc7e703fce812f81b28.js:1 GET https://partner.szesum.com/customLoginPage?uid=sUujKVZrjUm-Xe8Qq_uj-&action=signin&error=Invalid%20username%20or%20password. net::ERR_FAILED
```

It seems to be a regression since we had seen it worked before. But we are not sure.

**Comment by jim-king-2000** [comment]
It turns out that we do not handle options method in our custom server. Next.js does the correct behavior.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
